ckanext/gitdatahub/plugin.py

```diff
@@ -77,18 +78,31 @@ def after_update(self, context, pkg_dict):
                 json.dumps(body, indent=2),
                 contents.sha
                 )
-            
+            try:
+                lfs_pointers = [obj.name for obj in repo.get_contents("data")]
+            except UnknownObjectException as e:
+                lfs_pointers = list()
+
             gitattributes_body = ''
-            for obj in body['resources']:
+            for index, obj in enumerate(body['resources']):
                 gitattributes_body += "data/{} filter=lfs diff=lfs merge=lfs -text\n".format(obj['title'])
-            
+                if obj['title'] not in lfs_pointers:
```

Comment:
We need to add an scenario where the object exists in the `lfs_pointers` but the `sha256` and `size` changed. (This means the data file in the resource was updated). So if `obj['title'] in lfs_pointers` and `sha256` is different, we need to update the lfs_pointer.